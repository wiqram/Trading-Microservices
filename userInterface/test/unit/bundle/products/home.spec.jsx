import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
import ProductHome from '../../../../src/bundle/product/home';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('ProductHome Snapshot Test', () => {
    const products = [{
        id: '1',
        productId: "SA001",
        productName: 'Current A/C',
        currentBalance: 1121.11,
        accountNumber: '122111',
        sortCode: '122212',
        availableBalance: 12212.12

    },
    {
        id: '2',
        productId: "CA001",
        productName: 'Saving A/C',
        currentBalance: 11.11,
        accountNumber: '111111',
        sortCode: '121212',
        availableBalance: 1212.12

    }]
    let store;
    it('renders correctly with all props given error', () => {
        store = mockStore({
            productState: {
                isLoading: false,
                error: 'Test',
                products: []
            }
        });
        const wrapper = create(<Provider store={store}>
            <BrowserRouter>
                <ProductHome />
            </BrowserRouter>
        </Provider>)
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly with all props given isLoading - false', () => {
        store = mockStore({
            productState: {
                isLoading: false,
                error: '',
                products: products
            }
        });
        const wrapper = create(<Provider store={store}>
            <BrowserRouter>
                <ProductHome />
            </BrowserRouter>
        </Provider>)
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with all props given isLoading - true', () => {
        store = mockStore({
            productState: {
                isLoading: true,
                error: '',
                products: []
            }
        });
        const wrapper = create(<Provider store={store}>
            <BrowserRouter>
                <ProductHome />
            </BrowserRouter>
        </Provider>)
        expect(wrapper).toMatchSnapshot();
    });
});