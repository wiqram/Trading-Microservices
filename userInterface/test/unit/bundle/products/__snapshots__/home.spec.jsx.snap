// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductHome Snapshot Test renders correctly with all props given error 1`] = `
<div
  className="styles__Alert-sc-1ldttm7-11 bxELWj"
>
  <div
    className="styles__Row-sc-1ldttm7-2 iMcGKE"
  >
    <strong>
      Test
    </strong>
  </div>
</div>
`;

exports[`ProductHome Snapshot Test renders correctly with all props given isLoading - false 1`] = `
<div
  className="styles__Container-sc-1ldttm7-0 chTCyR"
>
  <div
    className="styles__Row-sc-1ldttm7-2 style__Card-sc-1rp0p1n-0 kkqgzJ"
  >
    <div
      className="styles__Col-sc-1ldttm7-1 bTiKQz"
    >
      <div
        className="styles__Row-sc-1ldttm7-2 style__CardRow-sc-1rp0p1n-2 hSSDTv"
      >
        <div
          className="style__ProductName-sc-1rp0p1n-1 lgSegm"
        >
          Current A/C
        </div>
        <div
          className="style__AccountNumber-sc-1rp0p1n-3 ghNEoX"
        >
          122111/122212
        </div>
      </div>
      <div
        className="styles__Row-sc-1ldttm7-2 style__CardRow-sc-1rp0p1n-2 hSSDTv"
      >
        <div
          className="style__Currency-sc-1rp0p1n-5 XQCQa"
          size={36}
        >
          <div
            className="currency__TextStyle-sc-1uazvwe-0 kMHNch"
          >
            £1,121.11
          </div>
        </div>
        <div
          className="style__BalanceLabel-sc-1rp0p1n-4 gFnvle"
        >
          Current Balance
        </div>
      </div>
      <div
        className="styles__Row-sc-1ldttm7-2 style__CardRow-sc-1rp0p1n-2 hSSDTv"
      >
        <div
          className="style__Currency-sc-1rp0p1n-5 bqdoCa"
          size={22}
        >
          <div
            className="currency__TextStyle-sc-1uazvwe-0 kMHNch"
          >
            £12,212.12
          </div>
        </div>
        <div
          className="style__BalanceLabel-sc-1rp0p1n-4 gFnvle"
        >
          Available Balance
        </div>
      </div>
    </div>
    <div
      className="styles__Col-sc-1ldttm7-1 style__NavItems-sc-1rp0p1n-6 bozNBC"
    >
      <a
        href="/product/SA001"
        onClick={[Function]}
      >
        <button
          className="styles__Button-sc-1ldttm7-3 styles__ButtonPrimary-sc-1ldttm7-4 jkYupd"
        >
          More Details
        </button>
      </a>
    </div>
  </div>
  <div
    className="styles__Row-sc-1ldttm7-2 style__Card-sc-1rp0p1n-0 kkqgzJ"
  >
    <div
      className="styles__Col-sc-1ldttm7-1 bTiKQz"
    >
      <div
        className="styles__Row-sc-1ldttm7-2 style__CardRow-sc-1rp0p1n-2 hSSDTv"
      >
        <div
          className="style__ProductName-sc-1rp0p1n-1 lgSegm"
        >
          Saving A/C
        </div>
        <div
          className="style__AccountNumber-sc-1rp0p1n-3 ghNEoX"
        >
          111111/121212
        </div>
      </div>
      <div
        className="styles__Row-sc-1ldttm7-2 style__CardRow-sc-1rp0p1n-2 hSSDTv"
      >
        <div
          className="style__Currency-sc-1rp0p1n-5 XQCQa"
          size={36}
        >
          <div
            className="currency__TextStyle-sc-1uazvwe-0 kMHNch"
          >
            £11.11
          </div>
        </div>
        <div
          className="style__BalanceLabel-sc-1rp0p1n-4 gFnvle"
        >
          Current Balance
        </div>
      </div>
      <div
        className="styles__Row-sc-1ldttm7-2 style__CardRow-sc-1rp0p1n-2 hSSDTv"
      >
        <div
          className="style__Currency-sc-1rp0p1n-5 bqdoCa"
          size={22}
        >
          <div
            className="currency__TextStyle-sc-1uazvwe-0 kMHNch"
          >
            £1,212.12
          </div>
        </div>
        <div
          className="style__BalanceLabel-sc-1rp0p1n-4 gFnvle"
        >
          Available Balance
        </div>
      </div>
    </div>
    <div
      className="styles__Col-sc-1ldttm7-1 style__NavItems-sc-1rp0p1n-6 bozNBC"
    >
      <a
        href="/product/CA001"
        onClick={[Function]}
      >
        <button
          className="styles__Button-sc-1ldttm7-3 styles__ButtonPrimary-sc-1ldttm7-4 jkYupd"
        >
          More Details
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`ProductHome Snapshot Test renders correctly with all props given isLoading - true 1`] = `
<div
  className="styles__Row-sc-1ldttm7-2 iMcGKE"
>
  <div
    className="styles__Col-sc-1ldttm7-1 eeaKic"
  >
    <b>
      Loading...
    </b>
  </div>
</div>
`;
